src/scheduler.h

```diff
@@ -9,29 +9,33 @@
 
 static constexpr int32_t SCHEDULER_MINTICKS = 50;
 
+class SchedulerTask;
+
+using SchedulerTask_ptr = std::unique_ptr<SchedulerTask>;
+
 class SchedulerTask : public Task
 {
 public:
+	SchedulerTask(uint32_t delay, TaskFunc&& f) : Task(std::forward<TaskFunc>(f)), delay(delay) {}
+
 	void setEventId(uint32_t id) { eventId = id; }
 	uint32_t getEventId() const { return eventId; }
 
 	uint32_t getDelay() const { return delay; }
 
 private:
-	SchedulerTask(uint32_t delay, TaskFunc&& f) : Task(std::move(f)), delay(delay) {}
-
 	uint32_t eventId = 0;
 	uint32_t delay = 0;
 
-	friend SchedulerTask* createSchedulerTask(uint32_t, TaskFunc&&);
+	friend SchedulerTask_ptr createSchedulerTask(uint32_t delay, TaskFunc&& f);
```

Comment:
Not needed anymore I think, since the constructor is public

(boo `friend` is bad)